### PR TITLE
Convert from gflags to Abseil Flags library

### PR DESCRIPTION
**Pull request by michael-kernel-sanders**

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -19,6 +19,13 @@ workspace(name = "oak")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
 
+http_archive(
+    name = "com_google_absl",
+    urls = ["https://github.com/abseil/abseil-cpp/archive/27c30ec671cb7b5ba84c4e79feff7fd0b0ac6338.zip"],
```
Comment (tiziano88) [inline WORKSPACE]: Perhaps also add a comment similar to https://github.com/google/asylo/blob/master/asylo/bazel/asylo_deps.bzl#L152 , so we have a hint of roughly when to update (for lack of proper version numbers).

File: oak/server/oak_manager.h [hunk 2]
```diff
@@ -24,7 +26,7 @@ namespace oak {
 
 class OakManager final : public Manager::Service {
  public:
-  OakManager();
+  explicit OakManager(const std::string& enclave_path);
```
Comment (tiziano88) [inline oak/server/oak_manager.h]: Worth using `string_view` here? https://abseil.io/tips/1

File: WORKSPACE (hunk 1 above)
Comment (michael-kernel-sanders) [reply WORKSPACE]: Done.

File: oak/server/oak_manager.h (hunk 2 above)
Comment (michael-kernel-sanders) [reply oak/server/oak_manager.h]: It doesn't seem very useful with only one caller that we control, but sure why not...  :)

**Pull request merged**